mesonbuild/modules/gnome.py

```diff
@@ -1664,49 +1679,47 @@ def _make_mkenum_custom_target(
                                   # https://github.com/mesonbuild/meson/issues/973
                                   absolute_paths=True)
 
-    @permittedKwargs({'sources', 'prefix', 'install_header', 'install_dir', 'stdinc',
-                      'nostdinc', 'internal', 'skip_source', 'valist_marshallers',
-                      'extra_args'})
     @typed_pos_args('gnome.genmarshal', str)
-    def genmarshal(self, state: 'ModuleState', args: T.Tuple[str], kwargs) -> ModuleReturnValue:
+    @typed_kwargs(
+        'gnome.genmarshal',
+        DEPEND_FILES_KW,
+        INSTALL_KW.evolve(name='install_header'),
+        KwargInfo('build_always', bool, default=False),
```

Comment:
There's a bunch of mysterious kwargs here, that are not in the documentation *and* are not in permittedKwargs. Attempting to use them raises a fatal error, e.g. 

```
genmarshal/meson.build:1:0: ERROR: Got unknown keyword arguments "build_always"
```

In commit 80d665e8decd644d737930d450f0bc66d6b5b02c, we added that error checking in, by following the documentation. At the time, the error message would be:

```
Meson encountered an error in file genmarshal/meson.build, line 1, column 0:
Invalid keyword argument build_always.
```

We also have some terrible manual argument checking in known_kwargs and known_custom_target_kwargs, but they've been entirely overridden for 4.5 years now, and no one has ever complained that they need these undocumented kwargs, which, again, were undocumented.

I see no grounds to re-add support for them half a decade later, especially just to mark at least one of them as deprecated in a later commit in the same PR... and if any of them do get added, I want them to be documented.